iModelCore/ECDb/ECDb/ECSql/ECSqlFieldFactory.cpp

```diff
@@ -111,6 +111,38 @@ ECSqlColumnInfo ECSqlFieldFactory::CreateColumnInfoForProperty(ECSqlPrepareConte
             }
         }
 
+    if (propertyNameExp != nullptr && propertyNameExp->GetParent() != nullptr && propertyNameExp->GetParent()->GetParent() != nullptr && propertyNameExp->GetParent()->GetParent()->GetType() == Exp::Type::NavValueCreationFunc)
+        {
+        ECClassCR ecClass = *ctx.GetECDb().Schemas().GetClass(
+            propertyNameExp->GetClassRefExp()->GetAs<ClassNameExp>().GetSchemaName(),
+            propertyNameExp->GetClassRefExp()->GetAs<ClassNameExp>().GetClassName()
+        );
+        ECPropertyCP ecProperty = ecClass.GetPropertyP(propertyNameExp->GetPropertyPath()[0].GetName().c_str());
+        ECStructClassCP structType = nullptr;
+        DateTime::Info dateTimeInfo;
+        ECTypeDescriptor typeDescriptor = DetermineDataType(dateTimeInfo, structType, ctx.Issues(), *ecProperty);
+        ECSqlPropertyPath propertyPath;
+        if (isGenerated)
+            {
+            propertyPath.AddEntry(*generatedProperty);
+            rootClass = &generatedProperty->GetClass();
+            }
+        else
+            propertyPath.AddEntry(*ecProperty);
+        return ECSqlColumnInfo(

```

Comment:
@khanaffan can you review this piece? This is another parent parent parent situation, I hope there is a more robust way to do this.

Reply:
We don't need this piece of code, removed